portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,114 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
+  private final DatastoreService dataStore = DatastoreServiceFactory.getDatastoreService();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.setContentType("application/json");
+    Query query = new Query("Comment").addSort("likes", SortDirection.DESCENDING);
+
+    PreparedQuery comments = dataStore.prepare(query);
+    List<Comment> commentsToSet = new ArrayList<>();
+    for (Entity entity : comments.asIterable()) {
+        String user = (String) entity.getProperty("user");
+        String commentText = (String) entity.getProperty("comment");
+        long commentSize = commentText.length();
+        long time = (long) entity.getProperty("time");
+        Long likes = (Long) entity.getProperty("likes");
+        
+        long id = entity.getKey().getId();
+
+        Comment commentToSet = new Comment(commentText, user, commentSize, likes.intValue(), time, id);
+        commentsToSet.add(commentToSet);
+    }
+
+    Gson gson = new Gson();
+    String payload = gson.toJson(commentsToSet);
+
+    response.getWriter().println(payload);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    String commentText = request.getParameter("comment");
+    String user = request.getParameter("user");
+    long commentSize = commentText.length();
+    long time = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("user", user);
+    commentEntity.setProperty("comment", commentText);
+    commentEntity.setProperty("time", time);
+    commentEntity.setProperty("size", commentSize);
+    commentEntity.setProperty("likes", 0);
+
+    dataStore.put(commentEntity);
+
+    long id = commentEntity.getKey().getId();
+
+    Comment comment = new Comment(commentText, user, commentSize, 0, time, id);
+
+    String payload = new Gson().toJson(comment);
+    response.getWriter().println(payload);
+
+    response.sendRedirect("/index.html");
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Entity commentEntity;
+    Key key = KeyFactory.createKey("Comment", Long.parseLong(request.getHeader("id")));
+    
+    try {
+      commentEntity = dataStore.get(key);
+      commentEntity.setProperty("likes", ((Long)commentEntity.getProperty("likes")).intValue() + 1);
```

Comment:
Is `.intValue()` necessary here? I think you could just do 
```java
(Long)commentEntity.getProperty("likes") + 1L
```

(in Java, the way to make the literal value `1` into a `long` is to add the suffix `L`. `1` has type `int` but `1L` has type `long`)